lorawan-encoding/tests/lorawan.rs

```diff
@@ -651,36 +677,32 @@ fn test_join_accept_creator_with_cflist() {
 #[test]
 fn test_join_request_creator() {
     let buf = [0u8; 23];
-    let mut phy = JoinRequestCreator::with_options(buf, DefaultFactory).unwrap();
+    let mut phy = JoinRequestCreator::new(buf).unwrap();
     let key = [1; 16].into();
     phy.set_app_eui(&[0x04, 0x03, 0x02, 0x01, 0x04, 0x03, 0x02, 0x01])
         .set_dev_eui(&[0x05, 0x04, 0x03, 0x02, 0x05, 0x04, 0x03, 0x02])
         .set_dev_nonce(&[0x2du8, 0x10]);
 
-    assert_eq!(phy.build(&key), &phy_join_request_payload()[..]);
+    assert_eq!(phy.build(&key, &DefaultFactory), &phy_join_request_payload()[..]);
+}
+#[test]
+fn test_join_request_creator_long_buffer() {
+    let buf = [0u8; 255];
+    let mut phy = JoinRequestCreator::new(buf).unwrap();
+    let key = [1; 16].into();
+    phy.set_app_eui(&[0x04, 0x03, 0x02, 0x01, 0x04, 0x03, 0x02, 0x01])
+        .set_dev_eui(&[0x05, 0x04, 0x03, 0x02, 0x05, 0x04, 0x03, 0x02])
+        .set_dev_nonce(&[0x2du8, 0x10]);
+
+    assert_eq!(phy.build(&key, &DefaultFactory), &phy_join_request_payload()[..]);
```

Comment:
```suggestion
    assert_eq!(phy.build(&key, &DefaultFactory), &phy_join_request_payload());
```

Reply:
this needs to be compared against a slice. .as_slice() would be proper here but [..] is used everywhere